obspy/io/shapefile/core.py

```diff
@@ -0,0 +1,282 @@
+# -*- coding: utf-8 -*-
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import *  # NOQA
+from future.utils import native_str
+
+from obspy import Catalog
+from obspy.core.event import Origin, Magnitude
+from obspy.core.inventory import Inventory
+
+try:
+    import gdal
+    from osgeo import ogr, osr
+except ImportError as e:
+    has_GDAL = False
+    IMPORTERROR_MSG = str(e) + (
+        ". ObsPy's write support for shapefiles requires the 'gdal' module "
+        "to be installed in addition to the general ObsPy dependencies.")
+else:
+    has_GDAL = True
+    gdal.UseExceptions()
+
+
+def _write_shapefile(obj, filename, **kwargs):
+    """
+    Write :class:`~obspy.core.event.Catalog` object to a ESRI shapefile.
+
+    :type obj: :class:`~obspy.core.event.Catalog` or
+        :class:`~obspy.core.inventory.Inventory`
+    :param obj: ObsPy object for shapefile output
+    :type filename: str or file
```

Comment:
Yep, thanks. Can't work with a file-like object, as it always has to be multiple files by shapefile format definition..  Overlooked this one mention of it after copy/paste.